dbms/src/Flash/tests/gtest_topn_executor.cpp

```diff
@@ -75,6 +79,127 @@ class ExecutorTopNTestRunner : public DB::tests::ExecutorTest
+                        if (direction > 0)
+                        {
+                            for (size_t i = 0; i < limit; ++i)
+                            {
+                                ASSERT_EQ(col_0->getUInt(i), i + 1);
+                            }
+                        }
+                        else
+                        {
+                            for (size_t i = 0; i < limit; ++i)
+                            {
+                                ASSERT_EQ(col_0->getUInt(i), 6 - i);
+                            }
+                        }
+                    }
+            }
+        }
+    }
+    {
+        const ColumnsWithTypeAndName
+            ori_col
+            = {
+                toVec<Int64>(col_name[0], ColumnWithInt64{1, 0, -1}),
+                toVec<String>(col_name[1], ColumnWithString{"1   ", "1  ", "1 "}),
+                toVec<String>(col_name[2], ColumnWithString{"1", "2", "3"}),
+                toVec<UInt64>(col_name[3], ColumnWithUInt64{1, 1, 0}),
+            };
+        for (bool no_collation : {true, false})
+        {
+            Block block(ori_col);
+            SortDescription description;
+            size_t limit = ori_col[0].column->size();
+            const auto * collator_1_ptr = TiDB::ITiDBCollator::getCollator(TiDB::ITiDBCollator::BINARY);
+            const auto * collator_2_ptr = collator_1_ptr;
+            if (no_collation)
+            {
+                collator_1_ptr = nullptr;
+                collator_2_ptr = nullptr;
+            }
+            description.emplace_back(ori_col[1].name, 1, 0, collator_1_ptr);
+            description.emplace_back(ori_col[2].name, 1, 0, collator_2_ptr);
+            sortBlock(block, description, limit);
+            auto cols = block.getColumnsWithTypeAndName();
+            ASSERT_EQ(ori_col.size(), cols.size());
+            const auto * col_0 = typeid_cast<const ColumnInt64 *>(cols[0].column.get());
+            ASSERT_TRUE(col_0);
+            ASSERT_EQ(col_0->size(), limit);
+
+            for (int i = 0; i < int(limit); ++i)
+            {
+                ASSERT_EQ(col_0->getElement(i), i - 1);
+            }
+        }
+        {
+            Block block(ori_col);
+            SortDescription description;
+            size_t limit = ori_col[0].column->size();
+            description.emplace_back(ori_col[3].name, 1, 0, nullptr);
+            description.emplace_back(ori_col[0].name, 1, 0, nullptr);
+            sortBlock(block, description, limit);
+            auto cols = block.getColumnsWithTypeAndName();
+            ASSERT_EQ(ori_col.size(), cols.size());
+            const auto * col_0 = typeid_cast<const ColumnInt64 *>(cols[0].column.get());
+            ASSERT_TRUE(col_0);
+            ASSERT_EQ(col_0->size(), limit);
+
+            for (size_t i = 0; i < limit; ++i)
+            {
+                ASSERT_EQ(col_0->getElement(i), i - 1);
+            }
+        }
+    }
+}
+CATCH
```

Comment:
It does not look like an Executor Unit Test.
Maybe it should be a separate file under `Interpreters/tests`.

Reply:
Done